### PR TITLE
Fix misc voiceover issues (#1437)

### PR DESCRIPTION
**Pull request by roostr**
### First time contributor checklist

- [X] I have read the [README](https://github.com/WhisperSystems/Signal-iOS/blob/master/README.md) and [CONTRIBUTING](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md) documents
- [X] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)
(I signed it twice. The first time I didn't know to put my github username in Address #2)

### Contributor checklist

- [X] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [X] My commits are rebased on the latest master branch
- [X] My commits are in nice logical chunks
- [X] My contribution is fully baked and is ready to be merged as is
- [X] I have tested my contribution on these devices:
 * iPhone 6S, iOS 10.1.1
- [X] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

- - - - - - - - - -

### Description

Fixes issues from #1437. There was a dump of several things in that thread, and here's what's fixed:

1. In Voiceover mode, on the main listing of threads screen, you can no longer read the "there are no messages" view when it's hidden.
2. In Voiceover mode, in a two-party conversation, messages from yourself are now prepended with "Me: message" instead of "Other party's name: message" while reading the transcript.

There was an issue reported wherein some languages (like Italian) didn't read messages at all in the conversation thread, instead reading out "text_message_accessibility_label" for every chat bubble. That must be fixed in a third-party dependency, jessesquires/JSQMessagesViewController. An [issue was created](https://github.com/jessesquires/JSQMessagesViewController/issues/1903) and [a PR was submitted](https://github.com/jessesquires/JSQMessagesViewController/pull/1904).

I tested this issue by switching between Italian, English, and Croatian on my iPhone 6s running 10.1.1 and using Voiceover to read the related controls.

### CODE REVIEW

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m [hunk 1]
```diff
@@ -217,7 +217,8 @@ - (NSString *)senderId {
 }
 
 - (NSString *)senderDisplayName {
-    if (self.thread) {
+    BOOL isSenderMe = [_senderDisplayName isEqualToString:NSLocalizedString(@"ME_STRING", @"")];
```
Comment (michaelkirk) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: I think `[self.interaction isKindOfClass:[TSOutgoingMessage class]]` would be less brittle

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m [hunk 2]
```diff
@@ -217,7 +217,8 @@ - (NSString *)senderId {
...
+    if (self.thread && !isSenderMe) {
         return _thread.name;
```
Comment (michaelkirk) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: Can you verify that this works with messages in a group thread?

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m (hunk 1 above)
Comment (roostr) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: Certainly. Wasn't sure which between the two would be more robust.

Should I push --force the changed commits to this branch?

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m (hunk 2 above)
Comment (roostr) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: Yup, I'll look into it.

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: yes please.

**Comment by roostr** [comment]
@michaelkirk I pushed a new commit with a better approach. Now the adapter uses the contactsManager to look up the display name for the signal id at the time the model object is created, so the senderDisplayName property is setup correctly at creation time. No overriding of senderDisplayName is necessary anymore.

I believe this works for both contact messages and group messages (I changed both code paths), but I wasn't sure how best to set up a group message to test it with the staging server (I currently have only my own phone number authed on staging, and so far I've been testing my fixes in a chat with myself).

Could you give me guidance on an easy way to make a test group chat for testing?

**Comment by roostr** [comment]
Update: Confirmed that in a group message, it works perfectly now. Attributes it to the full name of the participant if it's an incoming message, and "Me" if it's outgoing.

**Comment by michaelkirk** [comment]
This looks good. Thanks for fixing this. 

I rebased, tested locally, and merged at 087f7539790514baf34e2c6fbfe0a2fd5b50dd17

**Comment by roostr** [comment]
Great, thanks for the review and merge.

**Pull request closed without merging**